redirector/main.go

```diff
@@ -1,48 +1,78 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"net/http"
 	"regexp"
 )
 
-type RedirectHandler func([]string) (string, error)
+type RawRedirect struct {
+	ShortUrl       string
+	LongUrl        string
+	PatternParams  []string
+	TemplateParams []string
+}
 
-type Redirector struct {
-	Pattern string
-	Handler RedirectHandler
+type Redirect struct {
+	ShortRegex    *regexp.Regexp
+	LongRegex     *regexp.Regexp
+	ShortTemplate string
+	LongTemplate  string
 }
 
-var redirectors = []Redirector{
-	Redirector{`/r([0-9a-fA-F]{6,40})/(\d+)/?`, HandleCommitComment},
-	Redirector{`/r([0-9a-fA-F]{6,40})/?`, MakeStaticRedirector("https://github.com/dolphin-emu/dolphin/commit/")},
-	Redirector{`/i(\d+)/?`, MakeStaticRedirector("https://code.google.com/p/dolphin-emu/issues/detail?id=")},
-	Redirector{`/i(\d+)/(\d+)/?`, HandleIssueComment},
-	Redirector{`/pr/?(\d+)/?`, MakeStaticRedirector("https://github.com/dolphin-emu/dolphin/pull/")},
-	Redirector{`/pr(/.*)?`, MakeStaticRedirector("https://github.com/dolphin-emu/dolphin/pulls")},
-	Redirector{`/dl(/.*)?`, MakeStaticRedirector("https://dolphin-emu.org/download/")},
-	Redirector{`/gh(/.*)?`, MakeStaticRedirector("https://github.com/dolphin-emu/dolphin")},
-	Redirector{`/git(/.*)?`, MakeStaticRedirector("https://github.com/dolphin-emu/dolphin")},
-	Redirector{`/faq(/.*)?`, MakeStaticRedirector("https://dolphin-emu.org/docs/faq/")},
-	Redirector{`/bbs(/.*)?`, MakeStaticRedirector("https://forums.dolphin-emu.org/")},
+var redirects = []Redirect{
+	Compile(RawRedirect{"/r%s/%s", "https://github.com/dolphin-emu/dolphin/commit/%s#commitcomment-%s", []string{`([0-9a-fA-F]{6,40})`, `(\d+)`}, []string{"$1", "$2"}}),
```

Comment:
Really needs some comments to explain wtf is going on with the %s and the $1 and the regexp parts and ...